### ISSUE TITLE
Add tests around application/vnd.api+json MIME type, review/update documentation

### ISSUE THREAD

**Issue by rthbound** [rails-api/active_model_serializers#1534]
Rails briefly included `application/vnd.api+json` as a default :json mime type, but that mime type has [now been removed](https://github.com/rails/rails/pull/23712/files).

@bf4 requested:
> would you mind opening an issue in ams, as well, so we can add a test there? Also to confirm docs are up to date.

@bf4 what exactly should we test? I'm guessing we'd want to ensure AMS will handle un-registering and re-registering the `application/vnd.api+json` MIME type when a user has chosen the `:json_api` adapter:
```
ActiveModelSerializers.config.adapter = :json_api
```

**Comment by bf4** [comment]
The philosophical issue is that we would never want a json:api response or request to return non-json:api data.  Specifically, the json adapter should be incompatible with a request containing json:api headers.  This means json api should probably have its own registered media type, renderer and parser

**Comment by bf4** [comment]
ref: sample repo to reproduce issue from https://github.com/rthbound/ams-json-api-broken-23712/commit/b27eaaf6c2bffc7e55c988ae0311fda31fe60bec#diff-f6d71e4e1f802b4480a42578418d49afR2 as mentioned in https://github.com/rails/rails/issues/23712

**Comment by rthbound** [comment]
@bf4 this issue more accurately descibes the rails issue, which has been resolved, https://github.com/rails/rails/issues/23766 and the minimal amount of work that was required to reproduce it.

Your [exchange with @ianks](https://github.com/rails/rails/pull/23712#issuecomment-185031053) led me to try a couple of, I guess erroneous, configuration options gleaned from a link you provided: [here](https://github.com/rails/rails/commit/6f73b1906fea9ac29f257e8e0c88c117901bbb9e#diff-6a80fa5836403509e86a81c2117af391R12) and [here](https://github.com/rails/rails/commit/6f73b1906fea9ac29f257e8e0c88c117901bbb9e#diff-384a5a15d8d53de799fb6541688ea5f9R117). Does the code you linked above have any bearing on this ticket's outcome? I don't understand your meaning.

**Comment by bf4** [comment]
My first take on the rails pr was to separate the jsonapi media type.  In the end, we chose to remove it, rather than have an incomplete implementation. See discussion there for the reasons.

Can you summarize your resolution and close this issue?

B mobile phone

**Comment by rthbound** [comment]
This issue was created so that we could add a test here, and also to confirm docs are up to date.

It would appear there's nothing to do, so I'll now close this issue.
